lib/openstack_query/handlers/presets/preset_handler_base.py

```diff
@@ -0,0 +1,154 @@
+        :param filter_func_kwargs: A dictionary of keyword: argument pairs to pass into filter function
+        """
+
+        filter_func = self._get_mapping(preset, prop)
+        if not filter_func:
+            raise QueryPresetMappingError(
+                "Preset Not Found: failed to find filter_function mapping for preset "
+                f"'{preset.name}' and property '{prop.name}'"
+                "does the preset work with property specified?"
+            )
+
+        try:
+            _ = self._check_filter_func(filter_func, filter_func_kwargs)
+        except TypeError as err:
+            raise QueryPresetMappingError(
+                f"Preset Argument Error: failed to build filter_function for preset '{preset.name}'"
+            ) from err
+
+        return lambda a: self._filter_func_wrapper(
+            a, filter_func, prop_func, filter_func_kwargs
+        )
+
+    @staticmethod
+    def _filter_func_wrapper(
+        item: Any,
+        filter_func: FilterFunc,
+        prop_func: PropFunc,
+        filter_func_kwargs: Optional[PresetKwargs] = None,
+    ):
+        """
+        Method that acts as a wrapper to the filter function, if the property cannot be found for the resource
+        we return False before calling the filter function - since there's no property to compare.
+        :param item: An openstack resource item
+        :param filter_func: A filter function to run if property given can be retrieved from openstack resource
+        :param **filter_func_kwargs: A dictionary of keyword args to pass to filter function
+        """
+        try:
+            item_prop = prop_func(item)
+        except AttributeError:
+            return False
+        if filter_func_kwargs:
+            return filter_func(item_prop, **filter_func_kwargs)
+        return filter_func(item_prop)
+
+    @staticmethod
+    def _check_filter_func(
+        func: FilterFunc, func_kwargs: Optional[PresetKwargs] = None
+    ) -> bool:
+        """
+        Method that checks a given function can accept a set of kwargs as arguments.
+        :param func: function to test
+        :param func_kwargs: kwargs to test
+        """
+
+        signature = inspect.signature(func)
+
+        # skip first parameter for filter function as it is a positional arg which takes the output
+        # of a property function and use that to compare against
+        parameters = list(signature.parameters.values())[1:]
+
+        has_varargs = any(param.kind == param.VAR_POSITIONAL for param in parameters)
+        has_varkwargs = any(param.kind == param.VAR_KEYWORD for param in parameters)
+
+        for param in parameters:
+            if param.kind == param.POSITIONAL_OR_KEYWORD:
+                param_name = param.name
+                if param_name in func_kwargs:
+                    kwargs_value = func_kwargs[param_name]
+                    param_type = param.annotation
+                    if not isinstance(kwargs_value, param_type):
```

Comment:
I agree we should fail noisily - however in a previous comment you mentioned for `check_kwarg_mapping` method you mentioned that it should return True/False and that False is okay 

I think we should check types and fail fast - otherwise the user will receive an error only after the query is run. In this case after the openstacksdk command is run - which could take 2-3 minutes.

Reply:
I altered the function to return a reason when it fails